### DISCUSSION TITLE
Media Extraction

### DISCUSSION THREAD

**Discussion by livrasand** [sircharlo/meeting-media-manager#386] [General]
One question, where do you extract the multimedia from? How do you do it? I would like to use the media extraction function.

**Comment by sircharlo** [comment]
I'm not sure I understand the question?

**Reply by livrasand** [reply]
How do you get the images and videos? Do you get them from JW.ORG or JWPUB files?

**Reply by livrasand** [reply]
I have an app, called Attendant Zoom, it is totally free. I would like to use your code to download the images and videos automatically, without the need for human intervention.

**Reply by sircharlo** [reply]
My code is open source, so no secrets there. What license is your app released under?

**Reply by livrasand** [reply]
My app (Attendant Zoom) is private code, but I can embed your code to download the media. I just can't find where you have the function, how does it work?

**Reply by sircharlo** [reply]
My code is licensed under the GNU AGPL v3 license. As such, I require that any app or software available for download that reuses part of my code also be open source. Which parts of my code were altered, if any, also needs to be clearly stated.

As such, you wouldn't be able to use any of my code in your closed-source app. Feel free to browse the code to get a feel for how it works, of course. But a direct copy paste would be a violation.

I'm sure you understand my reasons for licensing my code in this way. It's nothing personal; I just want there to be no ambiguity as to what's happening behind the scenes when media is fetched. Anyone can look at the source code and see exactly what's happening and how the code works.
